subsys/bluetooth/host/direction.c

```diff
@@ -155,6 +156,47 @@ static int hci_df_read_ant_info(uint8_t *switch_sample_rates,
 	return 0;
 }
 
+/* @brief Function handles send of HCI commnad to enable or disables CTE
+ *        transmission for given advertising set.
+ *
+ * @param[in] adv               Pointer to advertising set
+ * @param[in] enable            Enable or disable CTE TX
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static int hci_df_set_adv_cte_tx_enable(struct bt_le_ext_adv *adv,
+					bool enable)
+{
+	struct bt_hci_cp_le_set_per_adv_recv_enable *cp;
+	struct cmd_state_set state;
+	struct net_buf *buf;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_CL_CTE_TX_ENABLE, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+
+	cp->handle = adv->handle;
+	cp->enable = enable ? 1 : 0;
+
+	cmd_state_set_init(&state, adv->flags, BT_PER_ADV_CTE_ENABLED,
+			   enable);
+	cmd_data_state_set(buf, &state);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_SET_CL_CTE_TX_ENABLE,
+				   buf, NULL);
+
+	if (err) {
+		return err;
+	}
+
+	return 0;
```

Comment:
I was first going to point out to remove the empty line after assignment to `err` and checking its value, but if you're not going to extend this in the future you could just eliminate `err` completely and do `return bt_hci_cmd_send_sync(...);`

Reply:
Corrected according to suggestion